ion-schema/src/ion_path.rs

```diff
@@ -0,0 +1,90 @@
+use ion_rs::value::owned::Element;
+use ion_rs::value::Builder;
+use ion_rs::Symbol;
+use std::fmt;
+use std::fmt::Formatter;
+
+#[derive(Clone, PartialEq)]
+pub enum IonPathElement {
+    IndexedElement { index: usize },
```

Comment:
I think we can trim the `IndexedElement` variant name since the enum itself already says `Element`.

```suggestion
    Index { index: usize },
```